logplex_formatter.go

```diff
@@ -76,13 +77,25 @@ func NewLogplexBatchFormatter(b Batch, eData []errData, config *Config) HTTPForm
 // Request returns a properly constructed *http.Request, complete with headers
 // and ContentLength set.
 func (bf *LogplexBatchFormatter) Request() (*http.Request, error) {
-	req, err := http.NewRequest("POST", bf.stringURL, bf)
+	u, err := url.Parse(bf.stringURL)
```

Comment:
We parse this here, only to call String() and url.Parse again in extractCredentials.